stackslib/src/net/download/nakamoto/download_state_machine.rs

```diff
@@ -1553,6 +1564,24 @@ impl NakamotoDownloadStateMachine {
         }
     }
 
+    /// Find the highest available tenure ID.
+    /// Returns Some(consensus_hash) for the highest tenure available from at least one node.
+    /// Returns None if no tenures are available from any peer.
+    fn find_highest_available_tenure(&self) -> Option<ConsensusHash> {
+        let (t1, t2) = Self::find_unconfirmed_tenure_ids(
+            &self.wanted_tenures,
+            self.prev_wanted_tenures.as_ref().unwrap_or(&vec![]),
+            &self.available_tenures,
+        );
+        if let Some(ch) = t2 {
+            return Some(ch);
+        } else if let Some(ch) = t1 {
+            return Some(ch);
+        } else {
+            return None;
+        }
```

Comment:
Could be reduced to:

```suggestion
        t2.or(t1)
```

Reply:
Fixed